src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -66,6 +66,39 @@ def ffmpeg_read(bpayload: bytes, sampling_rate: int) -> np.array:
     return audio
 
 
+def apply_stride(tokens, stride):
+    max_token_n = tokens.shape[-1]
+    max_input_n = max(input_n for input_n, _, _ in stride)
+    ratio = max_token_n / max_input_n
+    for i, (input_n, left, right) in enumerate(stride):
+        token_n = int(round(input_n * ratio))
+        left_token = int(round(left / input_n * token_n))
+        right_token = int(round((input_n - right) / input_n * token_n))
+        # This is CTC to preseve decoding, we need to duplicate
+        # next letter, and last letter
+
+        first_letter = tokens[i, left_token]
+        tokens[i, :left_token] = first_letter
+
+        last_letter = tokens[i, right_token - 1]
+        tokens[i, right_token:] = last_letter
+
+
+def chunk_iter(inputs, feature_extractor, chunk_len, stride_left, stride_right):
+    inputs_len = inputs.shape[0]
+    step = chunk_len - stride_left - stride_right
```

Comment:
What does stride mean in this context? Because step doesn't equal stride_left + stride_right or just one of them, it is kind of hard to interpret what exactly stride_left or stride_right represent.

Reply:
If you calculate the step in this way, the actual overlap with the left chunk would be 2 * (stride_left + stride_right).
Consider, for example, chunk_len = 15, stride_left = 3, stride_right = 3, and step = 15-3-3 = 9.
0 chunk: start=0, end=15,
1 chunk: start=9, end=24,
2 chunk: start=18, end=33

For 1 chunk, the overlap with the 0 chunk is 6, and the overlap with the 2 chunk is 6, totaling 12 of overlap, but the intended overlap was 3 with the left chunk and 3 with the right chunk, totaling 5 of overlap.